### ISSUE TITLE
Buffer Overflow airdecap-ng

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#949]
**Reported by adam on 8 Nov 2011 22:56 UTC**

Hey There,

Using airdecap-ng today when trying to run it with a capture file I get the following output;

```
adam@VGN:~$ airdecap-ng -e "PNET PCS ONLY" -b 00:19:5B:EC:F1:68 -p l3tsgr0w_l3tsgr0w_l3tsgr0w_ capture-01.cap
*** buffer overflow detected ***: airdecap-ng terminated
======= Backtrace: =========
/lib/libc.so.6(__fortify_fail+0x50)[Memory map: ========
00110000-00245000 r-xp 00000000 08:01 2097275    /lib/libcrypto.so.0.9.8
00245000-0024d000 r--p 00134000 08:01 2097275    /lib/libcrypto.so.0.9.8
0024d000-0025c000 rw-p 0013c000 08:01 2097275    /lib/libcrypto.so.0.9.8
0025c000-0025f000 rw-p 00000000 00:00 0 
0038f000-003a9000 r-xp 00000000 08:01 2097231    /lib/libgcc_s.so.1
003a9000-003aa000 r--p 00019000 08:01 2097231    /lib/libgcc_s.so.1
003aa000-003ab000 rw-p 0001a000 08:01 2097231    /lib/libgcc_s.so.1
004a3000-004bf000 r-xp 00000000 08:01 2097393    /lib/ld-2.12.1.so
004bf000-004c0000 r--p 0001b000 08:01 2097393    /lib/ld-2.12.1.so
004c0000-004c1000 rw-p 0001c000 08:01 2097393    /lib/ld-2.12.1.so
005c5000-005d8000 r-xp 00000000 08:01 2097346    /lib/libz.so.1.2.3.4
005d8000-005d9000 r--p 00012000 08:01 2097346    /lib/libz.so.1.2.3.4
005d9000-005da000 rw-p 00013000 08:01 2097346    /lib/libz.so.1.2.3.4
00859000-009b0000 r-xp 00000000 08:01 2097405    /lib/libc-2.12.1.so
009b0000-009b2000 r--p 00157000 08:01 2097405    /lib/libc-2.12.1.so
009b2000-009b3000 rw-p 00159000 08:01 2097405    /lib/libc-2.12.1.so
009b3000-009b6000 rw-p 00000000 00:00 0 
009ff000-00a43000 r-xp 00000000 08:01 2097689    /lib/libssl.so.0.9.8
00a43000-00a44000 ---p 00044000 08:01 2097689    /lib/libssl.so.0.9.8
00a44000-00a45000 r--p 00044000 08:01 2097689    /lib/libssl.so.0.9.8
00a45000-00a48000 rw-p 00045000 08:01 2097689    /lib/libssl.so.0.9.8
00c68000-00c69000 r-xp 00000000 00:00 0          [vdso](0x93e970]
/lib/libc.so.6(+0xe486a)[0x93d86a]
airdecap-ng[0x80490bb]
airdecap-ng[0x804a64b]
/lib/libc.so.6(__libc_start_main+0xe7)[0x86fce7]
airdecap-ng[0x8048ea1]
=======)
00ead000-00eaf000 r-xp 00000000 08:01 2100950    /lib/libdl-2.12.1.so
00eaf000-00eb0000 r--p 00001000 08:01 2100950    /lib/libdl-2.12.1.so
00eb0000-00eb1000 rw-p 00002000 08:01 2100950    /lib/libdl-2.12.1.so
08048000-08051000 r-xp 00000000 08:01 1452535    /usr/local/bin/airdecap-ng
08051000-08052000 r--p 00008000 08:01 1452535    /usr/local/bin/airdecap-ng
08052000-08053000 rw-p 00009000 08:01 1452535    /usr/local/bin/airdecap-ng
08053000-08094000 rw-p 00000000 00:00 0 
08d98000-08db9000 rw-p 00000000 00:00 0          [rw-p 00000000 00:00 0 
b7807000-b780b000 rw-p 00000000 00:00 0 
bfe3b000-bfe5c000 rw-p 00000000 00:00 0          [stack](heap]
b77fa000-b77fc000)
Aborted
```

Attention would be great, I have attached the capture file.

**Comment by aircrack-ng** [comment]
**Comment by adam on 8 Nov 2011 22:56 UTC**

Sorry, couldn't attach file, it was around 13mb in size.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 8 Nov 2011 22:56 UTC**

You can use http://demo.ovh.com/ and post the URL here.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 8 Nov 2011 22:56 UTC**

Fixed display.

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 8 Nov 2011 22:56 UTC**

This is likely the same bug #728 (ticket)

**Comment by aircrack-ng** [comment]
**Comment by misterx on 8 Nov 2011 22:56 UTC**

Fixed in aircrack-ng/aircrack-ng@3d140fed2c76dcb1b5c5c51ccb8753129fe3ac5e and aircrack-ng/aircrack-ng@d7e2399ea949e0e1227c23556710445d4c99fc14.
